ocaml/typing/solver_intf.mli

```diff
@@ -296,13 +298,25 @@ module type Solver_polarized = sig
 
   (** Generalizes all reachable variables whose level is above [current_level],
       whose value is fully determined, by putting their level to [generic_level].*)
-      val generalize_structure :
-      current_level:int ->
-      generic_level:int ->
-      'a obj ->
-      ('a, 'l * 'r) mode ->
-      log:changes ref option ->
-      unit
+  val generalize_structure :
+    current_level:int ->
+    generic_level:int ->
+    'a obj ->
+    ('a, 'l * 'r) mode ->
+    log:changes ref option ->
+    unit
+
+  (** Copies all reachable variables whose level is at or above [copy_from_level],
+      and optionally puts the level of the copy to [copy_to_level].
+      If no [copy_to_level] is given, the copy will have the same level as the original.
+      Returns the copied mode *)
+  val copy :
+    copy_scope:copy_scope ->
+    copy_from_level:int ->
+    ?copy_to_level:int ->
```

Comment:
Rather than have `copy_to_level` be optional, I think you can just pass `generic_level` or even `Sys.max_int` as it's value when using it to implement `duplicate`.